src/OrbitGl/App.cpp

```diff
@@ -940,17 +940,22 @@ void OrbitApp::Disassemble(uint32_t pid, const FunctionInfo& function) {
 }
 
 void OrbitApp::ShowSourceCode(const orbit_client_data::FunctionInfo& function) {
+  ScopedMetric metric{metrics_uploader_,
+                      orbit_metrics_uploader::OrbitLogEvent::ORBIT_SOURCE_CODE_SHOW};
```

Comment:
Good point! I will take a look

Reply:
I dont think I agree with this. There are actually 2 problems. First, this would be the only metric (so far) where we collect the time including the wait time for the user. Second, I just realized the user can also click the cancel button when the source code is not found (This happens in main window `ShowSourceCode`). In this case the metric currently also sends `Success`, but it should send status `Cancelled`. 

I recognize that this is becoming a bigger change that might require some refactoring in OrbitApp and OrbitMainWindow, but I think its worth it.

Reply:
Make sense! I will think about how to refactor the codes in `OrbitMainWindow` and `OrbitApp`